Fix post icons in Admin UI

morganestes opened the pull request:
Allows post icons that aren't Dashicons to be displayed properly in the Admin UI.

This changes the return value of `bapt_get_content_block_dashicon()` to indicate if a `$menu_icon` is an SVG string or a URL and returns that instead of the Dashicon. In the front end, check for these values and print the correct HTML `img` tags for each, with a fallback to Dashicons as currently used.

Fixes #8

morganestes committed:
Fix the post type icon to handle multiple types

Extends the `h2` to handle SVG or URLs instead of just Dashicons.

morganestes committed:
Changed return values for dashicons function

Added return values of 'url' and 'svg' to indicate if either of those
was used to register a menu_icon in a post type. Use that value to
trigger a conditional on the front end to display the correct icon.

wpmark commented on inc/bapt-functions.php:
```diff
@@ -9,14 +9,24 @@
 function bapt_get_content_block_dashicon( $post_type_obj ) {
 	if ( null !== $post_type_obj->menu_icon ) {
+
+		if ( strstr( $post_type_obj->menu_icon, 'data:image/svg+xml;' ) ) {
```
@morganestes could you add some comments here please as to what these if statements are doing

morganestes replied:
You bet. I'll update the PR.

morganestes committed:
Improve checking menu icon for SVG or URLs

Use `strpos()` to check for string matches as it's easier on memory,
and add inline docs to show what we're looking for in each one.

morganestes committed:
Update docs for dashicon checker function

Splits the docblock into a main and extra description,
with details about checking for inline SVG, URLs to icons, and
returning hints instead of icon names.

Also, add a hint to `register_post_type()` to find the properties
of the object we're comparing inside the function.

morganestes commented:
@wpmark I switched from `strstr()` to `strpos()` for checking the icon since it's supposed to be a bit faster, and updated the docs inline and in the docblock.
